### PR TITLE
Mpuncel/consul 0.7.3 upgrade

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -58,6 +68,131 @@
...
+			"Comment": "v1.4.20-8-ged981a1d",
+			"Rev": "ed981a1d5ee78d20547091d0697a711e5185d07f"
+		},
+		{
+			"ImportPath": "github.com/aws/aws-sdk-go/private/protocol/xml/xmlutil",
+			"Comment": "v1.4.20-8-ged981a1d",
+			"Rev": "ed981a1d5ee78d20547091d0697a711e5185d07f"
+		},
+		{
+			"ImportPath": "github.com/aws/aws-sdk-go/private/waiter",
```
Comment (mpuncel) [inline Godeps/Godeps.json]: we should probably get rid of our dependency on `github.com/hashicorp/consul`, seems like it's adding a lot of bloat for us

Comment (mpuncel) [inline Godeps/Godeps.json]: 😞

File: Rakefile [hunk 2]
```diff
@@ -113,27 +114,31 @@ task :sync_consul_deps do
       first_three_joined = first_three.join("/")
 
       # in case the consul dependency is new and not in $GOPATH, do a "go get" first
-      e "go get #{first_three_joined}/..."
+      begin
+        e "go get #{first_three_joined}/..."
+      rescue
+        # sometimes "go get" fails with "no buildable go source files" but
```
Comment (petertseng) [inline Rakefile]: that can probably be done by the `-d` flag, I think? "no buildable go source files" sounds like it's trying to build+install

File: pkg/store/consul/consulutil/consultest.go [hunk 3]
```diff
@@ -77,7 +78,17 @@ func NewFixture(t *testing.T) Fixture {
 	//
 	// config.ConsulConfig.RaftConfig.StartAsLeader = true
 
-	a, err := agent.Create(config, os.Stdout)
+	// mpuncel: don't really know what reloadCh so just drain everything from it
```
Comment (rudle) [inline pkg/store/consul/consulutil/consultest.go]: lol

File: Godeps/Godeps.json (hunk 1 above)
Comment (rudle) [inline Godeps/Godeps.json]: 🤢

File: pkg/store/consul/consulutil/consultest.go (hunk 3 above)
Comment (mpuncel) [inline pkg/store/consul/consulutil/consultest.go]: and then i realized i wasn't even passing the right channel

File: Rakefile [hunk 4]
```diff
@@ -113,27 +114,31 @@ task :sync_consul_deps do
...
+            Dir.chdir(p2_repo_dir) do
+              e "godep update #{first_three.join("/")}/..."
+            end
+
+            puts "Successfully repaired mismatch in #{package_name}: consul has #{consul_rev} p2 had #{our_rev}"
+            success_count += 1
           else
             puts "No action taken for #{package_name}: consul's version #{consul_rev} is older than P2's #{our_rev}"
             skipped_count += 1
-            next
```
Comment (petertseng) [inline Rakefile]: slightly hilarious and sad. I only noticed this after you pointed it out.

File: Godeps/Godeps.json (hunk 1 above)
Comment (bwester) [inline Godeps/Godeps.json]: If you want to go down that path, do it _before_ you upgrade this mess ;)

File: Rakefile (hunk 2 above)
Comment (mpuncel) [inline Rakefile]: yeah that looks like what I want!

**Pull request closed without merging**